.github/workflows/notebooks-digest-updater-downstream.yaml

```diff
@@ -0,0 +1,107 @@
+# The aim of this GitHub workflow is to update the params.env file with the latest digest.
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        required: true
+        description: "Provide the name of the branch you want to update ex master, rhods-x.xx etc"
+      release-n:
+        required: true
+        description: "Provide release N version of the notebooks ex 2023a"
+
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Install Skopeo CLI
+        shell: bash
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install skopeo
+
+      # Checkout the release branch
+      - name: Checkout release branch
+        uses: actions/checkout@v3
+        with:
+          ref:  ${{ inputs.branch }}
+
+      # Create a new branch
+      - name: Create a new branch
+        run: |
+         git checkout -b digest-updater
+         git push --set-upstream origin digest-updater
+
+  update-n-version:
+    needs: [ initialize ]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Configure Git
+        run: |
+         git config --global user.email "github-actions[bot]@users.noreply.github.com"
+         git config --global user.name "GitHub Actions"
+
+      # Get the latest weekly build commit hash: https://github.com/opendatahub-io/notebooks/commits/2023a
+      - name: Checkout upstream notebooks repo
+        uses: actions/checkout@v3
+        with:
+         repository: red-hat-data-services/notebooks
+         ref: release-${{ inputs.release-n }}
+      - name: Retrive latest weekly commit hash from the release branch
+        id: hash
+        shell: bash
+        run: |
+          echo "HASH=$(git rev-parse --short HEAD)" >> ${GITHUB_OUTPUT}
+
+      # Checkout the release branch to apply the updates
+      - name: Checkout release branch
+        uses: actions/checkout@v3
+        with:
+          ref: digest-updater
+
+      - name: Fetch digest, and update the param.env file
+        run: |
+              IMAGES=("odh-minimal-notebook-image-n" "odh-minimal-gpu-notebook-image-n" "odh-pytorch-gpu-notebook-image-n" "odh-generic-data-science-notebook-image-n" "odh-tensorflow-gpu-notebook-image-n" "odh-trustyai-notebook-image-n")
+              REGEXES=("v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}" "cuda-[a-z]+-minimal-[a-z0-9]+-[a-z]+-3.9-${{ inputs.release-n }}-\d{8}-${{ steps.hash.outputs.HASH }}" "v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}" \
+                       "v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}" "cuda-[a-z]+-tensorflow-[a-z0-9]+-[a-z]+-3.9-${{ inputs.release-n }}-\d{8}-${{ steps.hash.outputs.HASH }}" "v1-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}")
```

Comment:
from maintenance point of view, i probably will define them as 
```
IMAGES_MATCH=(
  ["odh-minimal-notebook-image-n", "v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}"]
  ["odh-minimal-gpu-notebook-image-n", "cuda-[a-z]+-minimal-[a-z0-9]+-[a-z]+-3.9-${{ inputs.release-n }}-\d{8}-${{ steps.hash.outputs.HASH }}"]
  ["odh-pytorch-gpu-notebook-image-n", "v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}"]
  ["odh-generic-data-science-notebook-image-n", "v2-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}"]
  ["odh-tensorflow-gpu-notebook-image-n", "cuda-[a-z]+-tensorflow-[a-z0-9]+-[a-z]+-3.9-${{ inputs.release-n }}-\d{8}-${{ steps.hash.outputs.HASH }}" ]
  ["odh-trustyai-notebook-image-n", "v1-${{ inputs.release-n }}-\d{8}+-${{ steps.hash.outputs.HASH }}"]
) 

for IMAGE in "${IMAGES_MATCH[@]}"
do
    image=${IMAGE[0]}
    regex=${IMAGE[1]}
    echo "DEBUG: ${image} and ${regex}"
    ......
done```

Reply:
I did not test my snippet, only think it is easier to read the code
I am fine with your implementation.